examples/svelte/basic/src/App.svelte

```diff
@@ -0,0 +1,143 @@
+<script lang="ts">
+  import './index.css'
+  import {
+    createTable,
+    getCoreRowModelSync,
+    createTableInstance,
+  } from '@tanstack/svelte-table'
+
+  type Person = {
+    firstName: string
+    lastName: string
+    age: number
+    visits: number
+    status: string
+    progress: number
+  }
+
+  const table = createTable().setRowType<Person>()
+
+  const defaultData: Person[] = [
+    {
+      firstName: 'tanner',
+      lastName: 'linsley',
+      age: 24,
+      visits: 100,
+      status: 'In Relationship',
+      progress: 50,
+    },
+    {
+      firstName: 'tandy',
+      lastName: 'miller',
+      age: 40,
+      visits: 40,
+      status: 'Single',
+      progress: 80,
+    },
+    {
+      firstName: 'joe',
+      lastName: 'dirte',
+      age: 45,
+      visits: 20,
+      status: 'Complicated',
+      progress: 10,
+    },
+  ]
+
+  const defaultColumns = [
+    table.createGroup({
+      header: 'Name',
+      footer: props => props.column.id,
+      columns: [
+        table.createDataColumn('firstName', {
+          cell: info => info.value,
+          footer: props => props.column.id,
+        }),
+        table.createDataColumn(row => row.lastName, {
+          id: 'lastName',
+          cell: info => info.value,
+          header: () => 'Last Name',
+          footer: props => props.column.id,
+        }),
+      ],
+    }),
+    table.createGroup({
+      header: 'Info',
+      footer: props => props.column.id,
+      columns: [
+        table.createDataColumn('age', {
+          header: () => 'Age',
+          footer: props => props.column.id,
+        }),
+        table.createGroup({
+          header: 'More Info',
+          columns: [
+            table.createDataColumn('visits', {
+              header: () => 'Visits',
+              footer: props => props.column.id,
+            }),
+            table.createDataColumn('status', {
+              header: 'Status',
+              footer: props => props.column.id,
+            }),
+            table.createDataColumn('progress', {
+              header: 'Profile Progress',
+              footer: props => props.column.id,
+            }),
+          ],
+        }),
+      ],
+    }),
+  ]
+
+  let data = defaultData
+  let columns = defaultColumns
+
+  const rerender = () => {
+    data = data
+  }
+
+  $: instance = createTableInstance(table, {
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModelSync(),
+  })
+</script>
+
+<div class="p-2">
+  <table>
+    <thead>
+      {#each $instance.getHeaderGroups() as headerGroup}
+        <tr>
+          {#each headerGroup.headers as header}
+            <th colSpan={header.colSpan}>
+              {header.isPlaceholder ? null : header.renderHeader()}
```

Comment:
This should instead be
```svelte
{#if !header.isPlaceholder}
  <svelte:component this={header.renderHeader()} />
{/if}
```
because `renderHeader()` might be rendering a custom component, so in the svelte adapter the `render` method returns the custom components with props passed to them and the plain values as a dummy component

Reply:
Thanks for your help.

though I get this error with that code

```Uncaught TypeError: Class constructor SvelteComponent cannot be invoked without 'new'```

Can't tell if its a config issue or not

Reply:
I think that this issue is caused due to transpiling the classes in svelte-table to functions in the bundle here:
```js
// source code
class WrapperComponent extends SvelteComponent {
   // rest of code here
}

// in bundle
function WrapperComp(options) {
  var _this;

  _this = _SvelteComponent.call(this) || this;
```
We'll need to update the rollup/babel config of svelte-table to fix it.

Reply:
#3907